sdcm/kafka/kafka_producer.py

```diff
@@ -0,0 +1,410 @@
+
+        self.kafka_addresses = None
+        if 'broker-list' in self.kafka_options:
+            self.kafka_addresses = self.kafka_options['broker-list'].split(',')
+        elif 'bootstrap-server' in self.kafka_options:
+            self.kafka_addresses = self.kafka_options['bootstrap-server'].split(',')
+        elif self.params.get('kafka_backend') == 'localstack':
+            self.kafka_addresses = ['localhost:9092']
+
+        self.schema_registry_url = None
+        if 'schema-registry' in self.kafka_options:
+            self.schema_registry_url = self.kafka_options['schema-registry']
+        elif self.params.get('kafka_backend') == 'localstack':
+            self.schema_registry_url = 'http://localhost:8081'
+
+        self.topic = self.kafka_options.get('topic')
+        self.key_schema = self.kafka_options.get('key-schema')
+        self.value_schema = self.kafka_options.get('value-schema')
+
+        if self.record_format == 'avro':
+            if not self.schema_registry_url:
+                raise ValueError("schema_registry_url is required for Avro record format.")
+            self.schema_registry_client = SchemaRegistryClient({'url': self.schema_registry_url})
+
+            self.key_serializer = AvroSerializer(
+                self.schema_registry_client, self.key_schema, lambda obj, ctx: obj,
+            ) if self.key_schema else None
+            self.value_serializer = AvroSerializer(
+                self.schema_registry_client, self.value_schema, lambda obj, ctx: obj,
+            ) if self.value_schema else None
+
+            self.key_deserializer = AvroDeserializer(self.schema_registry_client)
+            self.value_deserializer = AvroDeserializer(self.schema_registry_client)
+        else:
+            self.key_serializer = StringSerializer('utf_8')
+            self.value_serializer = StringSerializer('utf_8')
+            self.key_deserializer = StringDeserializer('utf_8')
+            self.value_deserializer = StringDeserializer('utf_8')
+
+    def generate_data_from_schema(self, schema: dict, seed: int) -> dict | int | float | str | bool | bytes | None:
+        rand_gen = random.Random(seed)
+        if schema['type'] in ('record', 'struct'):  # 'record' type of schema for AVRO record format, 'struct' for JSON
+            data = {}
+            for field in schema['fields']:
+                # 'name' is used in AVRO type records, 'field' in JSON ones
+                field_name = field.get('name') or field.get('field')
+                if field_name == 'id':
+                    data[field_name] = seed
+                else:
+                    data[field_name] = self.generate_field_value(field['type'], rand_gen)
+            return data
+        else:
+            return self.generate_field_value(schema['type'], rand_gen)
+
+    def generate_field_value(self, field_type: str, rand_gen: random.Random) -> int | float | str | bool | bytes | None:
+        type_mapping = {
```

Comment:
Maybe instead of mapping str to str, better map field_type to generator function? should be more performant and clearer.

Reply:
Changed to field_type:gen_function mapping